tests/models/qwen2_audio/test_modeling_qwen2_audio.py

```diff
@@ -0,0 +1,310 @@
+        audio1 = ffmpeg_read(requests.get(url1).content, sampling_rate=self.processor.feature_extractor.sampling_rate)
+        url2 = "https://qianwen-res.oss-cn-beijing.aliyuncs.com/Qwen2-Audio/audio/glass-breaking-151256.mp3"
+        audio2 = ffmpeg_read(requests.get(url2).content, sampling_rate=self.processor.feature_extractor.sampling_rate)
+        url3 = "https://qianwen-res.oss-cn-beijing.aliyuncs.com/Qwen2-Audio/audio/f2641_0_throatclearing.wav"
+        audio3 = ffmpeg_read(requests.get(url3).content, sampling_rate=self.processor.feature_extractor.sampling_rate)
+
+        inputs = self.processor(text=prompts, audios=[audio1, audio2, audio3], return_tensors="pt", padding=True)
+
+        output = model.generate(**inputs, max_new_tokens=32)
+
+        EXPECTED_DECODED_TEXT = [
+            "Detect the language and recognize the speech:mister quilter is the apostle of the middle classes and we are glad to welcome his gospel",
+            "Generate the caption in English: Glass is breaking.",
+            "Classify the human vocal sound to VocalSound in English: Throat clearing",
+        ]
+        self.assertEqual(
+            self.processor.batch_decode(output, skip_special_tokens=True),
+            EXPECTED_DECODED_TEXT,
+        )
+
+    @slow
+    def test_small_model_integration_test_multiturn(self):
+        # Let' s make sure we test the preprocessing to replace what is used
+        model = Qwen2AudioForConditionalGeneration.from_pretrained("Qwen/Qwen2-Audio-7B-Instruct")
+
+        messages = [
+            {"role": "system", "content": "You are a helpful assistant."},
+            {
+                "role": "user",
+                "content": [
+                    {
+                        "type": "audio",
+                        "audio_url": "https://qianwen-res.oss-cn-beijing.aliyuncs.com/Qwen2-Audio/audio/glass-breaking-151256.mp3",
+                    },
+                    {"type": "text", "text": "What's that sound?"},
+                ],
+            },
+            {"role": "assistant", "content": "It is the sound of glass shattering."},
+            {
+                "role": "user",
+                "content": [
+                    {
+                        "type": "audio",
+                        "audio_url": "https://qianwen-res.oss-cn-beijing.aliyuncs.com/Qwen2-Audio/audio/f2641_0_throatclearing.wav",
+                    },
+                    {"type": "text", "text": "How about this one?"},
+                ],
+            },
+        ]
+
+        formatted_prompt = self.processor.apply_chat_template(messages, add_generation_prompt=True)
+
+        audios = []
+        for message in messages:
+            if isinstance(message["content"], list):
+                for ele in message["content"]:
+                    if ele["type"] == "audio":
+                        audios.append(
+                            ffmpeg_read(
```

Comment:
Do we have any other way to read some audio for these tests @ylacombe? e.g. through a dataset

I don't think this is a good pattern for three reasons:
* Importing a utils from the pipelines is a sign that our code isn't well contained -- a model sits lower in abstraction that pipelines and shouldn't need to know about them at all. The utility should probably be moved 
* This function uses subprocesses - anything which forces us out of python land is a code smell
* This assumes ffmpeg is available in the environment. It should be protected with a `@require_ffmpeg` decorator

Reply:
We actually do, cf how it's done for Whisper: https://github.com/huggingface/transformers/blob/9451a385261b30e7319a2c93285ab76161e8c003/tests/models/whisper/test_modeling_whisper.py#L1861-L1866

Reply:
That said, the reasons you gave against this pattern make total sense. I agree we should remove this pattern, at least in the modeling test.

In terms of how to actually use the model - as shown in the `.md` and discussed [here](https://github.com/huggingface/transformers/pull/32137#discussion_r1699130692) - I actually think `ffmpeg_read` is a great method because it performs a few operation under the hood: reading from whatever source there is, resampling the waveform and outputting a fp32 numpy array.

Since `ffmpeg` is quite a common utility to have, I feel like the strongest argument against it is that it uses sub-processes.

 Could we show in the `.md` how to use the model:
1. without using `ffmpeg_read`: i.e we could use [scipy read method](https://docs.scipy.org/doc/scipy/reference/generated/scipy.io.wavfile.read.html#scipy.io.wavfile.read) or `torchaudio.load` and show how to resample the audio
2. using `ffmpeg_read`: to show how to use the model from a list of urls

WDYT @amyeroberts ?

Reply:
Hey, @ylacombe @amyeroberts I've removed `ffmpeg_read` and replaced it with `liborsa` to read the audio. Can you help me take a look again? see this [commit](https://github.com/huggingface/transformers/pull/32137/commits/8d029b808f9e7f23701782f26526f7f6208fd7ca)